exporter/azureblobexporter/exporter.go

```diff
@@ -16,36 +23,156 @@ import (
+			ID: azidentity.ClientID(e.config.Auth.ClientID),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to create user managed identity credential: %w", err)
+		}
+		e.client, err = azblob.NewClient(e.config.URL, cred, nil)
+		if err != nil {
+			return fmt.Errorf("failed to create client with user managed identity: %w", err)
+		}
+	default:
+		return fmt.Errorf("unsupported authentication type: %s", authType)
+	}
 	return nil
 }
 
+func (e *azureBlobExporter) generateBlobName(signal pipeline.Signal) (string, error) {
+	// Get current time
+	now := time.Now()
+
+	var format string
+	switch signal {
+	case pipeline.SignalMetrics:
+		format = e.config.BlobNameFormat.MetricsFormat
+	case pipeline.SignalLogs:
+		format = e.config.BlobNameFormat.LogsFormat
+	case pipeline.SignalTraces:
+		format = e.config.BlobNameFormat.TracesFormat
+	default:
+		return "", fmt.Errorf("unsupported signal type: %v", signal)
+	}
+	return fmt.Sprintf("%s_%d", now.Format(format), randomInRange(1, int(e.config.BlobNameFormat.SerialNumRange))), nil
+}
+
 func (e *azureBlobExporter) Capabilities() consumer.Capabilities {
 	return consumer.Capabilities{MutatesData: false}
 }
 
-func (e *azureBlobExporter) ConsumeMetrics(_ context.Context, _ pmetric.Metrics) error {
-	return nil
+func (e *azureBlobExporter) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	// Marshal the metrics data
+	data, err := e.marshaller.marshalMetrics(md)
+	if err != nil {
+		return fmt.Errorf("failed to marshal metrics: %w", err)
+	}
+
+	return e.consumeData(ctx, data, pipeline.SignalMetrics)
 }
 
-func (e *azureBlobExporter) ConsumeLogs(_ context.Context, _ plog.Logs) error {
-	return nil
+func (e *azureBlobExporter) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	// Marshal the logs data
+	data, err := e.marshaller.marshalLogs(ld)
+	if err != nil {
+		return fmt.Errorf("failed to marshal logs: %w", err)
+	}
+
+	return e.consumeData(ctx, data, pipeline.SignalLogs)
 }
 
-func (e *azureBlobExporter) ConsumeTraces(_ context.Context, _ ptrace.Traces) error {
+func (e *azureBlobExporter) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	// Marshal the trace data
+	data, err := e.marshaller.marshalTraces(td)
+	if err != nil {
+		return fmt.Errorf("failed to marshal traces: %w", err)
+	}
+
+	return e.consumeData(ctx, data, pipeline.SignalTraces)
+}
+
+func (e *azureBlobExporter) consumeData(ctx context.Context, data []byte, signal pipeline.Signal) error {
+	// Generate a unique blob name
+	blobName, err := e.generateBlobName(signal)
+	if err != nil {
+		return fmt.Errorf("failed to generate blobname: %w", err)
+	}
+
+	blobContentReader := bytes.NewReader(data)
+	_, err = e.client.UploadStream(ctx, e.config.Container.Traces, blobName, blobContentReader, nil)
+	if err != nil {
+		return fmt.Errorf("failed to upload traces data: %w", err)
+	}
+
+	e.logger.Debug("Successfully exported traces to Azure Blob Storage",
```

Comment:
Message leftover from refactor, remove signal type from message and add it as a log attribute?

Reply:
Making it an attribute will make querying the data easier

Reply:
updated and made signal as attribute